### PR TITLE
[SPARK-14510][MLLIB] Add args-checking for LDA and StreamingKMeans

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

add the checking for LDA and StreamingKMeans
## How was this patch tested?

manual tests

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala [hunk 1]
```diff
@@ -269,6 +270,8 @@ class LDA private (
    */
   @Since("1.3.0")
   def setCheckpointInterval(checkpointInterval: Int): this.type = {
+    require(checkpointInterval > 0,
+      s"Period between checkpoints must be positive but got ${checkpointInterval}")
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala]: The legal bounds of checkpointInterval is >= 1 or = -1, -1 means disable checkpoint.

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/StreamingKMeans.scala [hunk 2]
```diff
@@ -218,6 +218,15 @@ class StreamingKMeans @Since("1.2.0") (
    */
   @Since("1.2.0")
   def setInitialCenters(centers: Array[Vector], weights: Array[Double]): this.type = {
+    weights.foreach {
+      weight =>
+        require(weight >= 0,
+          s"Weight for each inital center must be nonnegative but got [${weights.mkString(" ")}]")
+    }
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/StreamingKMeans.scala]: require(weights.forall(_ >= 0.0), s"...")

**Comment by zhengruifeng** [comment]
@yanboliang Thanks a lot

**Comment by yanboliang** [comment]
LGTM cc @mengxr @jkbradley

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala [hunk 3]
```diff
@@ -269,6 +270,8 @@ class LDA private (
    */
   @Since("1.3.0")
   def setCheckpointInterval(checkpointInterval: Int): this.type = {
+    require(checkpointInterval == -1 || checkpointInterval > 0,
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala]: Is `-1` documented somewhere?

**Comment by mengxr** [comment]
@zhengruifeng 

* Please try making each PR minimal. For example, this PR could be split into two PRs, one for LDA and one for StreamingKMeans.
* Create a JIRA for non-trivial changes. This PR requires `checkpoingInterval >= -1`, which might affect existing user code unless we know that for any value `< -1` there would be an error, just at a later time. However, this is not apparent.

**Comment by zhengruifeng** [comment]
@mengxr Sorry, I will make sure that my other PRs have corresponding JIRA.

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala (hunk 3 above)
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala]: The is no document direct for -1.
I add -1 according to @yanboliang 's comments.

Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala]: It has been documented at [here](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/impl/PeriodicGraphCheckpointer.scala#L73), but I think it's better to also clarify at ```setCheckpointInterval```.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**